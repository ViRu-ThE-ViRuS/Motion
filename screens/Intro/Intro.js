import React, { Component } from 'react';
import {
    View,
    Text,
    StyleSheet,
} from 'react-native';

import SvgUri from 'react-native-svg-uri';

import PrimaryFlatButton from './../../components/Button/PrimaryFlatButton';

export default class Intro extends Component {
    static navigationOptions = {
        header: null,
    }

    startLogin() {
        this.props.navigation.navigate('login');

        // const resetAction = NavigationActions.reset({
        //     index: 0,
        //     actions: [
        //         NavigationActions.navigate({ routeName: 'intro' }),
        //     ],
        // });
        // this.props.navigation.dispatch(resetAction);
    }

    render() {
        return (
          <View style={styles.container}>
            <View style={styles.center}>
              <SvgUri
                        source={require('./../res/motion_logo.svg')}
                        height="150"
                        width="150"
                    />
              <Text style={styles.heading}>Motion</Text>
              <Text style={styles.subHeadingPrimary}>is a mobile platform that lets you create change in the world.</Text>
              <Text style={styles.subHeadingSecondary}>You can create movements, gather like minded people, and make the world a better place.</Text>
            </View>
            <PrimaryFlatButton text="Next" onPress={() => this.startLogin()} />
          </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        justifyContent: 'space-between',
        flex: 1,
    },
    heading: {
        fontSize: 35,
        color: '#000',
        fontFamily: 'SourceSansPro_Bold',
    },
    subHeadingPrimary: {
        fontSize: 20,
        color: '#222',
        fontFamily: 'SourceSansPro_Light',
        textAlign: 'center',
        marginLeft: 15,
        marginRight: 15,
        marginTop: 15,
    },
    subHeadingSecondary: {
        fontSize: 20,
        color: '#666',
        fontFamily: 'SourceSansPro_Light',
        textAlign: 'center',
        marginLeft: 15,
        marginTop: 15,
        marginRight: 15,
    },
    center: {
        marginTop: 150,
        justifyContent: 'center',
        alignItems: 'center',
    },
});
