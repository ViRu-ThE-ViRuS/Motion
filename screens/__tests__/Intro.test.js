import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Intro from './../Intro/Intro';

beforeEach(() => {
    global.fetch = jest.fn((props) => ({
        text: jest.fn(),
    }));
});

it('Intro screen renders correctly', () => {
    const mockNavigation = {
        navigate: jest.fn((route) => true),
    };

    const tree = renderer.create(
      <Intro navigation={mockNavigation} />
    );

    expect(tree).toMatchSnapshot();
    expect(tree.getInstance().startLogin()).toBe(undefined);
});
