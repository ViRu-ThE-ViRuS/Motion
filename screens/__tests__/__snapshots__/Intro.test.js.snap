// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro screen tests renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 150,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../screens/res/mlogo.png",
        }
      }
      style={
        Object {
          "height": 150,
          "width": 150,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "SourceSansPro_Bold",
          "fontSize": 35,
        }
      }
    >
      Motion
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#222",
          "fontFamily": "SourceSansPro_Light",
          "fontSize": 20,
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 15,
          "textAlign": "center",
        }
      }
    >
      is a mobile platform that lets you create change in the world.
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#666",
          "fontFamily": "SourceSansPro_Light",
          "fontSize": 20,
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 15,
          "textAlign": "center",
        }
      }
    >
      You can create movements, gather like minded people, and make the world a better place.
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#5C7D87",
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontFamily": "SourceSansPro_Regular",
            "fontSize": 15,
          }
        }
      >
        Next
      </Text>
    </View>
  </View>
</View>
`;
